src/core/providers/device-provider/device.model.ts

```diff
@@ -0,0 +1,9 @@
+export interface DeviceModel {
+  isDevice: boolean;
+}
+
+export const createInitialDevice = (): DeviceModel => ({ isDevice: false });
+
+export interface DeviceContextModel {
+  isDevice: boolean;
+}
```

Comment:
We can do two things here:

Rename `isDevice` to `isTabletOrMobileDevice`

Or we can create a type:

```diff
type Device = 'mobile' | 'limitedTable' | 'DesktopOrExtendedTablet'
```